sdk/samples/iot/sample_pnp_thermostat_component.c

```diff
@@ -44,42 +47,61 @@ static const az_span max_temp_reported_property_name
 static const char iso_spec_time_format[] = "%Y-%m-%dT%H:%M:%S%z";
 
 static az_result build_command_response_payload(
-    sample_pnp_thermostat_component* thermostat_component,
-    az_json_writer* json_builder,
+    const sample_pnp_thermostat_component* thermostat_component,
     az_span start_time_span,
     az_span end_time_span,
-    az_span* response_payload)
+    az_span payload,
+    az_span* out_payload)
 {
   double avg_temp = thermostat_component->device_temperature_avg_total
       / thermostat_component->device_temperature_avg_count;
 
   // Build the command response payload
-  AZ_RETURN_IF_FAILED(az_json_writer_append_begin_object(json_builder));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(json_builder, report_max_temp_name_span));
+  az_json_writer json_builder;
```

Comment:
nit: while you are changing this, maybe rename this local name to `json_writer` or `jw`